src/runtime_src/hip/api/hip_device.cpp

```diff
@@ -48,6 +49,9 @@ device_init()
       continue;
     auto dev = std::make_shared<xrt::core::hip::device>(i);
     device_cache.add(i, std::move(dev));
+    auto mem_pool = std::make_shared<xrt::core::hip::memory_pool>(device_cache.get_or_error(i), MAX_MEMORY_POOL_SIZE_NPU, MEMORY_POOL_BLOCK_SIZE_NPU);
```

Comment:
So the mem_pool is shared with whom?  I am suspicious about shared ownership, but I do see the mem_pool pushed to a _db and a map, but is there one of these who is the owner and the other one an observer?   In other words do one outlive the other?    

Same comment for device, which seems to be owned by the device_cache, which should then be unique, but again, I am not sure because the code is not commented.